packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -376,13 +376,12 @@ function useFocusContainment(scopeRef: RefObject<Element[] | null>, contain?: bo
     };
 
     let onBlur = (e) => {
-      // Firefox doesn't shift focus back to the Dialog properly without this
+      // Use raf to delay long enough that subdialogs will fully unmount and not move focus away from parent menu's hovered item that triggered the close
```

Comment:
this went from being specific to dialogs to specific to menus with subdialogs?

Reply:
Yeah, there is quite some history with this stuff haha. So 4 years ago we had these issues: [fixing refocusing Dialog on underlay click for Firefox](https://github.com/adobe/react-spectrum/pull/378#top) + [Fix(#1168): Check document.activeElement in focus containment onBlur so focus in iframes is retained](https://github.com/adobe/react-spectrum/pull/1188#top), where essentially there were a variety of issues with retaining focus within a Dialog especially in Firefox. 

The first issue can be compare/contrasted in Firefox via [this build](https://reactspectrum.blob.core.windows.net/reactspectrum/72de2712609173cbb1a3633bd09018ee4bd0c6ab/storybook/index.html?path=/story/dialog--default) which predates all of the above changes (comes from https://github.com/adobe/react-spectrum/pull/321) and the current PR build. To reproduce, keyboard focus a dialog button -> click on the underlay -> hit Enter and see if the previously focused button is triggered. The second issue I don't have a build link reproducing the original issue since that iframe story was added with the original fix but you can just verify that clicking in the search field of the wikipedia iframe works and doesn't cause focus to be lost.

As for the current comment, the raf is still needed for subdialogs in menus specifically now it seems so I updated the comment.